cpp/modmesh/transform/fourier.hpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <modmesh/math.hpp>
+#include <modmesh/buffer/buffer.hpp>
+
+namespace modmesh
+{
+
+namespace transform
+{
+
+namespace detail
+{
+
+size_t bit_reverse(size_t n, const size_t bits);
+
+} /* end namespace detail */
+
+// TODO: The template of template is too complicate, we should find a way to make it easier.
```

Comment:
I left a TODO here to mention that we need to find a way to make it easier.

Reply:
This is informative.  Thanks.